Add customer cart and cart items API endpoints

SamehOssama opened the pull request:
Create API for CRUD operations on cart items by customer id
Update cart items model

SamehOssama committed:
Add cart and cart items basic API endpoints

SamehOssama committed:
Update cart item model properties and unique constraint

SamehOssama committed:
Add authentication to cart and cart items API endpoints

abdlrhman08 commented on stats/serializers.py:
```diff
@@ -7,7 +7,12 @@ class CartItemListCreateSerializer(serializers.ModelSerializer):
-        read_only_fields = ["id"]
+        read_only_fields = ["id", "customer"]
+
+    def create(self, validated_data):
+        validated_data["customer"] = self.context["request"].user.customer
+        obj = CartItem.objects.create(**validated_data)
+        return obj
 
```
Try as much as you can not to change any data in `create()`, Instead, override the customer field and make it a `SerializerMethodField`

See: https://github.com/abdlrhman08/Distributed-project-backend/blob/0168aabe2dedfa95bae83924f38be3be363e5839/stats/serializers.py#L10-L35

abdlrhman08 reviewed (changes requested):
Small edits

abdlrhman08 commented on stats/serializers.py:
```diff
@@ -0,0 +1,22 @@
+from .models import CartItem
+
+
+class CartItemListCreateSerializer(serializers.ModelSerializer):
+    class Meta:
+        model = CartItem
+        fields = ["id", "customer", "product", "quantity"]
+        read_only_fields = ["id", "customer"]
+
+    def create(self, validated_data):
+        validated_data["customer"] = self.context["request"].user.customer
+        obj = CartItem.objects.create(**validated_data)
+        return obj
+
+
+class CartItemUpdateDeleteSerializer(serializers.ModelSerializer):
+    class Meta:
+        model = CartItem
+        fields = ["id", "customer", "product", "quantity"]
+        read_only_fields = ["id", "customer", "product"]
```
We can merge them into a single serializer

SamehOssama committed:
Merge branch 'main' into add/cart-api

SamehOssama committed:
Add customer authentication for cart API endpoints

SamehOssama committed:
Add cart owner check authentication for cart API endpoints

SamehOssama committed:
Add a validator for cart product quantity

SamehOssama committed:
Removed update delete serializer temporarily

SamehOssama committed:
Fix cart items post endpoint to not require customer id

SamehOssama committed:
Re-enabled update serializer

SamehOssama committed:
Refactored quantity validator

SamehOssama committed:
Update cart item owner check for cart API endpoints

SamehOssama committed:
Formatting

SamehOssama committed:
Merge branch 'main' into add/cart-api

SamehOssama committed:
Update cart and cart items API views

Update delete and update cart API endpoint
Merged delete cart and list / create cart item endpoints
